xarray/plot/dataarray_plot.py

```diff
@@ -1848,9 +1848,10 @@ def _center_pixels(x):
         # missing data transparent.  We therefore add an alpha channel if
         # there isn't one, and set it to transparent where data is masked.
         if z.shape[-1] == 3:
-            alpha = np.ma.ones(z.shape[:2] + (1,), dtype=z.dtype)
+            safe_dtype = np.promote_types(z.dtype, np.uint8)
```

Comment:
The dtype should allow at least 0 and 255. But it will be converted in `np.ma.concatenate` at some point anyway so I thought it's best to just figure it out early and initialize correctly.